program/src/state/rules.rs

```diff
@@ -293,28 +306,36 @@ impl Rule {
                 );
 
                 // Negate the result.
-                (!result.0, result.1)
+                match result {
+                    Success(err) => Failure(err),
+                    Failure(err) => Success(err),
+                    Invalid(err) => Invalid(err),
+                }
             }
             Rule::AdditionalSigner { account } => {
                 msg!("Validating AdditionalSigner");
                 if let Some(signer) = accounts.get(account) {
-                    (signer.is_signer, self.to_error())
+                    if signer.is_signer {
+                        Success(self.to_error())
+                    } else {
+                        Failure(self.to_error())
+                    }
                 } else {
-                    (false, RuleSetError::MissingAccount.into())
+                    Failure(RuleSetError::MissingAccount.into())
```

Comment:
I think one should be `Invalid` instead of `Failure`.

Reply:
I think that results in an issue if you have AdditionalSigner in an OR statement, though. Because if you're not using the AdditionalSigner branch then it'll be Invalid with the addtional account.

Reply:
Ok maybe we can chat offline about this because it seems fine to return an Error/Invalid when the account is not found.  It won't roll up as such in an OR statement unless every OR'd rule was Error/Invalid.  That seems the same as how we are treating missing payload and `is_zeroed` failures?